### PR TITLE
TracedLayer Error Message Enhancement

### PR DESCRIPTION
**Pull request by zhhsplendid**
### PR types

New features
### PR changes

APIs
### Describe

Enhance TracedLayer Error Message

Note: this PR uses `assert` to check type somewhere and `check_type` somewhere, the reason is that the `check_type` skips checking when it is under dygraph mode.

### CODE REVIEW

**Review by Aurelius84 — DISMISSED** [review]

File: python/paddle/fluid/dygraph/jit.py [hunk 1]
```diff
@@ -43,10 +44,13 @@ def create_program_from_desc(program_desc):
...
     if isinstance(inputs, Variable):
         result_list.append(inputs)
-
-    if isinstance(inputs, (list, tuple)):
+    elif isinstance(inputs, (list, tuple)):
         for var in inputs:
             _extract_vars(var, result_list)
+    else:
+        raise TypeError(
+            "The type of 'each element of inputs' in fluid.dygraph.jit.TracedLayer.trace must be fluid.Variable, but received {}.".
```
Comment (swtkiwi) [inline python/paddle/fluid/dygraph/jit.py:52]: 自2.0起，所有Variable不再对用户暴漏，全部使用“Tensor”哈～辛苦改下

File: python/paddle/fluid/dygraph/jit.py [hunk 2]
```diff
@@ -1063,7 +1067,8 @@ def trace(layer, inputs):
 
         Args:
             layer (dygraph.Layer): the layer object to be traced.
-            inputs (list(Variable)): the input variables of the layer object.
+            inputs (list(Variable)|tuple(Variable)|Variable): the input
+                variables of the layer object.
```
Comment (swtkiwi) [inline python/paddle/fluid/dygraph/jit.py:1072]: Variable to “Tensor”

Comment (zhhsplendid) [reply python/paddle/fluid/dygraph/jit.py:1072]: Done.

File: python/paddle/fluid/dygraph/jit.py (hunk 1 above)
Comment (zhhsplendid) [reply python/paddle/fluid/dygraph/jit.py:52]: 这里的是代码报错，当前代码还是Variable类型，恐怕不适合更改，你看这里还用Variable可以吗？

**Review by swtkiwi — APPROVED** [review]
LGTM

**Pull request merged**